src/snowflake/connector/auth.py

```diff
@@ -664,3 +673,24 @@ def build_temporary_credential_name(host, user, cred_type):
     return "{host}:{user}:{driver}:{cred}".format(
         host=host.upper(), user=user.upper(), driver=KEYRING_DRIVER_NAME, cred=cred_type
     )
+
+
+def get_token_from_private_key(
```

Comment:
Can you add tests for this method?

Reply:
test added in test_key_pair_authentication.py